std/assembly/typedarray.ts

```diff
@@ -8,6 +8,14 @@ export class Int8Array extends TypedArray<i8,i32> {
   subarray(begin: i32 = 0, end: i32 = 0x7fffffff): Int8Array {
     return changetype<Int8Array>(super.subarray(begin, end));
   }
+  map(callbackfn: (value: i8, index: i32, array: Int8Array) => i8): Int8Array {
+    var length = this.length;
+    var result = new Int8Array(this.length);
+    for (let i = 0; i < length; ++i) {
+      result.__unchecked_set(i, callbackfn(this.__unchecked_get(i), i, this));
```

Comment:
This can also be written as `unchecked(result[i] = callbackfn(this.__unchecked_get(i), i, this))`, which will ensure that it still works whenever indexed access becomes more optimized or changes otherwise.

Reply:
Thanks dcode, but when I try to use it, it errors about the type because it doesn't have an index attribute.

![image](https://user-images.githubusercontent.com/3761339/49181472-71008880-f325-11e8-9dea-ea9391b905b2.png)

Reply:
Might be a TS error that results from the .d.ts not having it, but should compile with AS.

Reply:
I tried compiling with AS and it fails when trying to run the tests.

Reply:
For reference, the low-level way to write this is:

```ts
import { HEADER_SIZE } from "./internal/arraybuffer";
```

```ts
  map(callbackfn: (value: i8, index: i32, array: Int8Array) => i8): Int8Array {
    var length = this.length;
    var result = new Int8Array(length);
    var src = changetype<usize>(this.buffer) + HEADER_SIZE + this.byteOffset;
    var dst = changetype<usize>(result.buffer) + HEADER_SIZE;
    for (let i = 0; i < length; ++i) {
      store<i8>(dst + (i << alignof<i8>()), callbackfn(load<i8>(src + (i << alignof<i8>())), i, this));
      // can actually skip alignof<T> for i8/u8 because it's 0 anyway, but added it for clarity
    }
    return result;
  }
```